services/test_results.py

```diff
@@ -408,6 +408,8 @@ def should_write_flaky_detection(repoid: int, commit_yaml: UserYaml) -> bool:
 
 
 def should_read_flaky_detection(repoid: int, commit_yaml: UserYaml) -> bool:
-    return FLAKY_TEST_DETECTION.check_value(
-        identifier=repoid, default=False
-    ) and read_yaml_field(commit_yaml, ("test_analytics", "flake_detection"), False)
+    return (
+        FLAKY_TEST_DETECTION.check_value(identifier=repoid, default=False)
+        and read_yaml_field(commit_yaml, ("test_analytics", "flake_detection"), False)
+        or FLAKY_SHADOW_MODE.check_value(identifier=repoid, default=False)
```

Comment:
If there's some problem that we want to turn off displaying the results to users, then which flag would we want to flip?

Reply:
Turning off `FLAKY_SHADOW_MODE` would also disable writing. So, I think we might want another flag to just disable the reads.

Reply:
i updated the changes such that FLAKY_SHADOW_MODE retains it's current functionality on the write path and we're going to use FLAKY_DETECTION to toggle the beta, so we can separately disable the read path without affecting the write path